### PR TITLE
Zenodo crawler private dataset refactor

### PR DESCRIPTION
**Pull request by joeyzhou98**
## Description
We used to crawl private datasets on Zenodo by appending the dataset token in the URL which leaks it to the public in the git annex history. Now we use bearer token to authenticate via `DATALAD_ZENODO_token` environment variable which is more secure since it doesn't leak the token in the commit history

### CODE REVIEW

**Comment by cmadjar** [comment]
@joeyzhou98 looks like the tests are broken in this PR and I am not sure to understand why the openpain datasets show up as broken here as well (I see the same result in the PERFORM PR BTW).  

Currently, the only two datasets that are broken are: PERFORM and refseq_Homo_sapiens.

**Comment by joeyzhou98** [comment]
@cmadjar I don't think the problem comes from this PR, but my guess is that someone merged something in CONP-PCNO:conp-dataset that broke those tests because this PR doesn't change any of the PERFORM nor refseq_Homo_sapiens submodules. Maybe @mathdugre knows more

**Comment by joeyzhou98** [comment]
Oh wait, I know why the PERFORM dataset is failing. I had to recrawl the PERFORM dataset for security issues so I deleted the old repo and the test is trying to install the old one. This also explains why the PERFORM tests aren't failing for https://github.com/CONP-PCNO/conp-dataset/pull/629 but are failing for this PR. Notice the url is different for the PERFORM dataset submodule: https://github.com/CONP-PCNO/conp-dataset/pull/629/files#diff-fe7afb5c9c916e521401d3fcfb4277d5071798c3baf83baf11d6071742823584L28-R150. Now as for refseq, that is outside my control. The PERFORM tests should work once https://github.com/CONP-PCNO/conp-dataset/pull/629 is merged

**Comment by mathdugre** [comment]
> I deleted the old repo and the test is trying to install the old one.

I agree that this is what is causing the test failure for both the *main* branch and this PR.

As for the `refseq_Homo_sapiens` dataset, this issue seems  to be recurrent; around every 3 months (#552, #474, #402, #323). I'm not sure how this used to be solve, maybe @emmetaobrien has some insight on this.
Note: This should be fixed in another PR though

**Comment by cmadjar** [comment]
@mathdugre @joeyzhou98 Indeed, @emmetaobrien will fix the issue for the refseq dataset, see https://github.com/CONP-PCNO/conp-dataset/issues/638 

Thank you for the clarification for the PERFORM dataset. Makes sense.

File: scripts/Crawlers/ZenodoCrawler.py [hunk 1]
```diff
@@ -412,3 +362,53 @@ def get_readme_content(self, dataset):
...
+        with open(zenodo_config_path, "w") as f:
+            f.write(
+                """[provider:ZENODO]
+url_re = .*zenodo\\.org.*
+authentication_type = bearer_token
+credential = ZENODO
+
+[credential:ZENODO]
+# If known, specify URL or email to how/where to request credentials
+# url = ???
```
Comment (cmadjar) [inline scripts/Crawlers/ZenodoCrawler.py:395]: Shouldn't `url` be set here?

Comment (joeyzhou98) [reply scripts/Crawlers/ZenodoCrawler.py:395]: Nope, it is only needed when requesting credentials

Comment (cmadjar) [reply scripts/Crawlers/ZenodoCrawler.py:395]: right! Thank you :)

**Review by cmadjar — APPROVED** [review]
LGTM! Thank you @joeyzhou98

**Pull request merged**